internal/services/kusto/kusto_attached_database_configuration_resource.go

```diff
@@ -154,6 +154,27 @@ func resourceKustoAttachedDatabaseConfiguration() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if !features.FivePointOh() {
+		resource.Schema["cluster_id"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ExactlyOneOf: []string{"cluster_id", "cluster_resource_id"},
+			ValidateFunc: commonids.ValidateKustoClusterID,
+		}
+		resource.Schema["cluster_resource_id"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ForceNew:     true,
```

Comment:
Similar to what Steph said here, we'll have to remove `ForceNew` here as well for when people move from `cluster_resource_id` to `cluster_id` as `cluster_resource_id` will have to be removed which will cause `ForceNew` to trigger